tests/test_search_rollup.py

```diff
@@ -0,0 +1,285 @@
+import functools
```

Comment:
I'll go ahead and add some tests that target NULL values. That seems pretty important to validate.